rust/frontend/src/server.rs

```diff
@@ -317,11 +386,22 @@ async fn get_user_identity(
     Ok(Json(server.auth.get_user_identity(&headers).await?))
 }
 
-#[derive(Deserialize, Debug)]
+#[derive(Deserialize, Debug, ToSchema)]
 struct CreateTenantPayload {
     name: String,
 }
 
+/// Creates a new tenant.
+#[utoipa::path(
+    post,
+    path = "/api/v2/tenants",
+    request_body = CreateTenantPayload,
+    responses(
+        (status = 200, description = "Tenant created successfully", body = CreateTenantResponse),
+        (status = 401, description = "Unauthorized", body = ErrorResponse),
```

Comment:
are we forced to manually specify the error type this way? It would be nice if it could somehow be inferred off the return type.

Reply:
Working on defining common error types, which should simplify this: https://docs.rs/utoipa-gen/latest/utoipa_gen/derive.IntoResponses.html

Reply:
Ok - not sure what you have in mind but might talk to @Sicheng-Pan and @codetheweb the first as I know they thought a bit about the current error types.

Reply:
Just posted about this here: https://github.com/chroma-core/chroma/pull/3815#discussion_r1964574258

Summary is that the IntoResponses standard errors aren't compatible with our pattern. So, we'll stick with the standard ErrorResponse that the frontend already uses for now.